lib/gpw/client.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module GPW
+  class Client
+    def fetch_companies_list
+      response = fetch_gpw_stock_list
+      @companies = parse_data(response)
+      @companies.each { |company| Company.find_or_create_by!(company) }
+    end
+
+    private
+
+    def fetch_gpw_stock_list
+      conn_gpw = Faraday.new(
+        url: 'https://gpw.notoria.pl/widgets/ta/symbols.php',
+        params: {
+          callback: 'jQuery221045406664420430665_1647857422814',
+          _: '1647857422815',
+        },
+        headers: {
+          'Accept' => 'application/json',
+          'Accept-Language' => 'en-US,en;q=0.5',
+          'Content-Type' => 'application/json',
+          'X-Requested-With' => 'XMLHttpRequest',
+          'Connection' => 'keep-alive',
+          'Referer' => 'https://gpw.notoria.pl/widgets/ta/index.html?symbid=PLPKN0000018&symbname=PKNORLEN&symburl=/widgets/ta/symbols.php&dataurl=/widgets/ta/data.php&periods=1d,1w,1m&periods_intraday=1&ranges=1d,1m,3m,6m,1y,2y,3y,5y,10y',
+          'Sec-Fetch-Dest' => 'empty',
+          'Sec-Fetch-Mode' => 'cors',
+          'Sec-Fetch-Site' => 'same-origin',
+          'DNT' => '1',
+          'Sec-GPC' => '1',
```

Comment:
perhaps not all of these is required to make a successful call? you have a lot of things copied directly from the sample request from the browser that are not valuable in the calls from non-browser clients